tests/kernel/sched/schedule_api/prj.conf

```diff
@@ -2,6 +2,6 @@ CONFIG_ZTEST=y
 CONFIG_IRQ_OFFLOAD=y
 CONFIG_NUM_PREEMPT_PRIORITIES=30
 #CONFIG_SCHED_SCALABLE=y
-CONFIG_MAX_THREAD_BYTES=4
+CONFIG_MAX_THREAD_BYTES=6
```

Comment:
This looks suspicious.  You aren't creating any new `struct k_thread` objects, just re-using ones that already exist.  So the total number of thread kobj indexes shouldn't be changing.  If it is, you're probably re-using a thread that hasn't aborted yet (though IIRC that should be detected and fail), or there's a bug somewhere else.  Why do you need this setting?

Reply:
In this testcase, I want to creat 4 thread like following
{ K_PRIO_PREEMPT(0), K_MSEC(10) } ->
{ K_PRIO_PREEMPT(0), K_MSEC(20) } ->
{ K_PRIO_PREEMPT(1), K_MSEC(10) } ->
{ K_PRIO_PREEMPT(1), K_MSEC(20) }
the sanitycheck would have a compile mistake and the build.log file told me to change the macro CONFIG_MAX_THREAD_BYTES value, it should be 5 not 6.

Reply:
OK, we need to figure this out.  @andrewboie is the expert here.  But my read of the code is that the kobj indexes for those objects should have been freed for reuse by their previous thread aborts.  If that didn't happen, we either have a bug where the test is re-using threads incorrectly or thread abort isn't doing the right object free.

Reply:
if you are getting a **compiler** error, this isn't a bug with re-use of thread indexes, or indeed a runtime issue at all.

what is happening is that the entire zephyr kernel binary is searched for `struct k_thread` objects, and the ones found at build time get an index assigned to them, up to `CONFIG_MAX_THREAD_BYTES * 8`

so if you get a a build error that CONFIG_MAX_THREAD_BYTES is too small, then what has happened is that the number of `struct k_thread` objects found by the build exceeds the number of bits available defined by CONFIG_MAX_THREAD_BYTES.

This change would imply that a value of 4 (32 thread objects) was not enough. You can see what objects the build has found by running make with `VERBOSE=1`.

It does look like this test creates a lot of objects, without this PR the verbose output is:

```
gen_kobject_list.py: symbol 'tstacks' at 0x12c000 contains 10 object(s)
gen_kobject_list.py: symbol 'sema2' at 0x1890d0 contains 1 object(s)
gen_kobject_list.py: symbol 'sema3' at 0x1890e8 contains 1 object(s)
gen_kobject_list.py: symbol 't' at 0x124000 contains 10 object(s)
gen_kobject_list.py: symbol 'tdata' at 0x124600 contains 1 object(s)
gen_kobject_list.py: symbol 'end_sema' at 0x12546c contains 1 object(s)
gen_kobject_list.py: symbol 'tdata' at 0x1246a0 contains 1 object(s)
gen_kobject_list.py: symbol 'tthread' at 0x124820 contains 3 object(s)
gen_kobject_list.py: symbol 't' at 0x124780 contains 1 object(s)
gen_kobject_list.py: symbol 'pend_sema' at 0x189088 contains 1 object(s)
gen_kobject_list.py: symbol 'timer_sema' at 0x189100 contains 1 object(s)
gen_kobject_list.py: symbol 'timer' at 0x124740 contains 1 object(s)
gen_kobject_list.py: symbol 'sema' at 0x1890a0 contains 1 object(s)
gen_kobject_list.py: DIE DW_TAG_variable, size=14, has_children=False
    |DW_AT_name        :  AttributeValue(name='DW_AT_name', form='DW_FORM_string', value=b't', raw_value=b't', offset=49715)
    |DW_AT_decl_file   :  AttributeValue(name='DW_AT_decl_file', form='DW_FORM_data1', value=7, raw_value=7, offset=49717)
    |DW_AT_decl_line   :  AttributeValue(name='DW_AT_decl_line', form='DW_FORM_data1', value=124, raw_value=124, offset=49718)
    |DW_AT_decl_column :  AttributeValue(name='DW_AT_decl_column', form='DW_FORM_data1', value=18, raw_value=18, offset=49719)
    |DW_AT_type        :  AttributeValue(name='DW_AT_type', form='DW_FORM_ref4', value=7095, raw_value=7095, offset=49720)
    |DW_AT_location    :  AttributeValue(name='DW_AT_location', form='DW_FORM_exprloc', value=[145, 164, 124], raw_value=[145, 164, 124], offset=49724)

gen_kobject_list.py: File '/home/apboie/projects/zephyr/zephyr/tests/kernel/sched/schedule_api/src/test_sched_timeslice_reset.c', line 124:
gen_kobject_list.py:     kernel object 't' found on stack
gen_kobject_list.py: symbol 't' at 0x124a40 contains 10 object(s)
gen_kobject_list.py: symbol 'sema1' at 0x1890b8 contains 1 object(s)
gen_kobject_list.py: symbol 'user_thread' at 0x125040 contains 1 object(s)
gen_kobject_list.py: symbol 'user_sem' at 0x189118 contains 1 object(s)
gen_kobject_list.py: symbol 'ustack' at 0x168000 contains 1 object(s)
gen_kobject_list.py: symbol 'fdtable_lock' at 0x189030 contains 1 object(s)
gen_kobject_list.py: symbol '__device_sys_init_z_clock_driver_init0' at 0x189020 contains 1 object(s)
gen_kobject_list.py: symbol 'ztest_thread' at 0x125240 contains 1 object(s)
gen_kobject_list.py: symbol 'cpuhold_sem' at 0x1254d4 contains 1 object(s)
gen_kobject_list.py: symbol 'ztest_thread_stack' at 0x16e000 contains 1 object(s)
gen_kobject_list.py: symbol 'uart_ns16550_driver_api' at 0x1109dc contains 1 object(s)
gen_kobject_list.py: symbol '__device_uart_ns16550_0' at 0x189010 contains 1 object(s)
gen_kobject_list.py: symbol '__device_uart_ns16550_1' at 0x189000 contains 1 object(s)
gen_kobject_list.py: symbol 'z_main_stack' at 0x17a000 contains 1 object(s)
gen_kobject_list.py: symbol 'z_main_thread' at 0x125380 contains 1 object(s)
gen_kobject_list.py: symbol 'z_idle_threads' at 0x1252e0 contains 1 object(s)
gen_kobject_list.py: symbol 'z_idle_stacks' at 0x174000 contains 1 object(s)
gen_kobject_list.py: symbol 'z_interrupt_stacks' at 0x180000 contains 1 object(s)
gen_kobject_list.py: DIE DW_TAG_variable, size=17, has_children=False
    |DW_AT_name        :  AttributeValue(name='DW_AT_name', form='DW_FORM_strp', value=b'dummy_thread', raw_value=16052, offset=302356)
    |DW_AT_decl_file   :  AttributeValue(name='DW_AT_decl_file', form='DW_FORM_data1', value=1, raw_value=1, offset=302360)
    |DW_AT_decl_line   :  AttributeValue(name='DW_AT_decl_line', form='DW_FORM_data2', value=458, raw_value=458, offset=302361)
    |DW_AT_decl_column :  AttributeValue(name='DW_AT_decl_column', form='DW_FORM_data1', value=18, raw_value=18, offset=302363)
    |DW_AT_type        :  AttributeValue(name='DW_AT_type', form='DW_FORM_ref4', value=796, raw_value=796, offset=302364)
    |DW_AT_location    :  AttributeValue(name='DW_AT_location', form='DW_FORM_exprloc', value=[145, 220, 126], raw_value=[145, 220, 126], offset=302368)

gen_kobject_list.py: File '/home/apboie/projects/zephyr/zephyr/kernel/init.c', line 458:
gen_kobject_list.py:     kernel object 'dummy_thread' found on stack
gen_kobject_list.py: symbol 'offload_sem' at 0x189070 contains 1 object(s)
gen_kobject_list.py: symbol 'pipe_async_msgs' at 0x18904c contains 1 object(s)
gen_kobject_list.py: device instance at 0x189020 has no associated subsystem
gen_kobject_list.py: found 60 kernel object instances total
```

@andyross at least in the current version of this PR, some new static thread objects are introduced in test_thread_priority.c:

```
static struct k_thread tdata_prio[THREAD_NUM];
```

If we were right at the limit beforehand, this could push the necessary amount of thread bytes from 4 to 5 (which the PR currently does, don't know why it was increased to 6 in an earlier version)

tl;dr: I think this is OK.

Reply:
@andyross Hi, andyross.  It seems that andrewboie has made it clear why the system prompts to modify the configuration file.